public/js/icinga/events.js

```diff
@@ -172,6 +174,18 @@
             icinga.destroy();
         },
 
+        onVisibilityChange: function (event) {
+            var icinga = event.data.self.icinga;
+
+            if !!(document.visibilityState === undefined || document.visibilityState === 'visible') {
+                icinga.logger.info('Page visible, enabling auto-refresh');
+                icinga.loader.enableAutorefresh()
```

Comment:
The call to `enableAutorefresh()` enables auto refresh even if it was disabled before using the console. We could introduce a new property `autorefreshSuspended` which is updated internally by the `onVisibilityChange` handler. The loader could then decide whether auto refresh is enabled by taking the properties `autorefreshEnabled` and `autorefreshSuspended` into account. I'm open to other suggestions though.

Reply:
For me the use case "some guy fiddled with the console" too exotic to handle it.

If you want you can take it from here ...